util/chunk/row_container_test.go

```diff
@@ -204,3 +205,54 @@ func (r *rowContainerTestSuite) TestSortedRowContainerSortSpillAction(c *check.C
 	err = rc.Reset()
 	c.Assert(err, check.IsNil)
 }
+
+func (r *rowContainerTestSerialSuite) TestActionBlocked(c *check.C) {
+	sz := 4
+	fields := []*types.FieldType{types.NewFieldType(mysql.TypeLonglong)}
+	rc := NewRowContainer(fields, sz)
+
+	chk := NewChunkWithCapacity(fields, sz)
+	for i := 0; i < sz; i++ {
+		chk.AppendInt64(0, int64(i))
+	}
+	var tracker *memory.Tracker
+	var err error
+	// Case 1
+	tracker = rc.GetMemTracker()
+	tracker.SetBytesLimit(1450)
+	ac := rc.ActionSpill()
+	tracker.FallbackOldAndSetNewAction(ac)
+	for i := 0; i < 10; i++ {
+		err = rc.Add(chk)
+		c.Assert(err, check.IsNil)
+	}
+
+	ac.cond.L.Lock()
+	for ac.cond.status == notSpilled ||
+		ac.cond.status == spilling {
+		ac.cond.Wait()
+	}
+	ac.cond.L.Unlock()
+	ac.cond.L.Lock()
+	c.Assert(ac.cond.status, check.Equals, spilledYet)
+	ac.cond.L.Unlock()
+	c.Assert(tracker.BytesConsumed(), check.Equals, int64(0))
+	c.Assert(tracker.MaxConsumed(), check.Greater, int64(0))
+	c.Assert(rc.GetDiskTracker().BytesConsumed(), check.Greater, int64(0))
+
+	// Case 2
```

Comment:
What does `case 2` mean? There is no check logic for this case.

Reply:
Test action can block when spilling.  I add a check logic for this case. PTAL again.